### PR TITLE
Issue #2375: Add a view and administrative pages for directly managing files.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes backdrop/backdrop-issues#2375.

Replaces PR https://github.com/backdrop/backdrop/pull/1658.

### CODE REVIEW

**Comment by quicksketch** [comment]
Reopening to get a new sandbox.

**Comment by herbdool** [comment]
@quicksketch Looks pretty good. 

I don't think this is an issue, but I noticed the difference in the confirmation pages--where deleting an individual file lists each content that references it but when deleting multiple it just lists the files and the usage numbers. Seems that it has to be that way to prevent a potentially huge list.

File: core/modules/file/file.field.inc [hunk 1]
```diff
@@ -234,7 +234,10 @@ function file_field_insert($entity_type, $entity, $field, $instance, $langcode,
 
   // Add a new usage of each uploaded file.
   foreach ($items as $item) {
-    file_usage_add(file_load($item['fid']), 'file', $entity_type, $id);
+    $file = file_load($item['fid']);
+    if ($file) {
```
Comment (jenlampton) [inline core/modules/file/file.field.inc:238]: Can we do this in one line?
` if ($file = file_load($item['fid'])) {`

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/1711 where I'm adding tests.

**Pull request closed without merging**